source/opt/copy_prop_arrays.cpp

```diff
@@ -269,7 +269,7 @@ CopyPropagateArrays::BuildMemoryObjectFromExtract(
     // Convert the indices in the extract instruction to a series of ids that
     // can be used by the |OpAccessChain| instruction.
     for (uint32_t i = 1; i < extract_inst->NumInOperands(); ++i) {
-      uint32_t index = extract_inst->GetSingleWordInOperand(1);
+      uint32_t index = extract_inst->GetSingleWordInOperand(i);
```

Comment:
oh, subtle :)  Do you have a test case for this or did you catch it by visual inspection?

Reply:
We did not have a test case, but I added one.  I was looking at that code because I wanted to do something similar in the new pass, and noticed the error.